### PR TITLE
GEN-2432 - refact(ProductItem): make ProductItemV2 more reusable

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Moved _Widget_ Product Item component into `components/ProductItemV2`.  `ProductItem.tsx` doesn't get show as a renaming in git because I have changed some things on it. Will be marking then as comments but the most important one is that it now receives a `children` so we always show common things like _pillow_, _details_ and _documents_ and receive what's especfic from the consumer like _InputDay_ field.

## Justify why they are needed

For the new [checkout page](https://www.figma.com/design/p1TEzLLVqJB9gKDNTDVGU4/Hedvig.com?node-id=14-263&t=z9B58dtJeUsjHBlJ-0) we'll be using the same _Product Item_ card we're currently using for _widget_:

image

This PR is about making some changes on that component by extracting _Widget_ specific things from it so we can reuse most of it on the new checkout page.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4475?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4475** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4475?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4469** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4469?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/ProductItemV2/ProductItem.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+type Props = {
+  selectedOffer: Offer
+  children?: ReactNode
+  greenVariant?: boolean
+  defaultExpanded?: boolean
+  onDelete?: (event: React.MouseEvent<HTMLButtonElement>) => void
+  onEdit?: () => void
+  className?: string
+}
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductItemV2/ProductItem.tsx:34]: Before we're getting thinks like `mode` (`view`or `edit`) and not receiving `children`

File: apps/store/src/components/ProductItemV2/ProductItem.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  const toggleExpandCard = () => {
+    setExpanded((prev) => !prev)
+  }
+
+  const handleDelete = (event: MouseEvent<HTMLButtonElement>) => {
+    // Prevents expanding the card when clicking the delete button
+    event.stopPropagation()
+    props.onDelete?.(event)
+  }
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductItemV2/ProductItem.tsx:69]: Before, `event.stopPropagation()` had to be done inside `props.onDelete`. For convenience I'm adding here so if consumer forgot to add it inside `props.onDelete` we'd still be preventing card get's expanded when _delete_ button is clicked.

File: apps/store/src/components/ProductItemV2/ProductItem.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+          <div className={cardHeaderRow}>
+            <Text as="p" size="md" color="textTranslucentPrimary">
+              {props.selectedOffer.product.displayNameFull}
+            </Text>
+
+            {props.onDelete && (
+              <button className={deleteButton} onClick={handleDelete}>
+                <CrossIconSmall color={theme.colors.textSecondary} />
+              </button>
+            )}
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductItemV2/ProductItem.tsx:93]: This thing I'll be changing in the next PR but HTML doesn't allow `button` be descendent of `button` so I've noticed an error when this get's rendered.

File: apps/store/src/components/ProductItemV2/ProductItem.tsx (hunk 2 above)
Comment (alebedev) [reply apps/store/src/components/ProductItemV2/ProductItem.tsx:69]: Sounds like a good general rule - if `stopPropagation / preventDefault` is need, the very first level of event handlers must call it

File: apps/store/src/components/ProductItemV2/ProductItem.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    props.onDelete?.(event)
+  }
+
+  const pillow = props.selectedOffer.product.pillowImage
+
+  return (
+    <div className={clsx(card, props.greenVariant && cardGreenVariant, props.className)}>
+      <button
+        className={cardHeader}
+        style={{ gridTemplateColumns: pillow.src ? 'auto 1fr' : '1fr' }}
```
Comment (alebedev) [inline apps/store/src/components/ProductItemV2/ProductItem.tsx:77]: when would we not have pillow and should we have a special style for it?

File: apps/store/src/components/ProductItemV2/ProductItem.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+              </button>
+            )}
+          </div>
+          <Text as="p" color="textTranslucentSecondary">
+            {props.selectedOffer.exposure.displayNameShort}
+          </Text>
+        </div>
+      </button>
+
+      <Space y={1}>
```
Comment (alebedev) [inline apps/store/src/components/ProductItemV2/ProductItem.tsx:101]: Would be good to remove Space and SpaceFlex in some future PR

**Review by alebedev — APPROVED** [review]
Looks good

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Jul 12, 5:44 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4475).

**Pull request merged**